src/unix/async.c

```diff
@@ -129,18 +120,7 @@ static void uv__async_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   }
 
   wa = container_of(w, struct uv__async, io_watcher);
-
-#if defined(__linux__)
-  if (wa->wfd == -1) {
-    uint64_t val;
-    assert(n == sizeof(val));
-    memcpy(&val, buf, sizeof(val));  /* Avoid alignment issues. */
-    wa->cb(loop, wa, val);
-    return;
-  }
-#endif
-
-  wa->cb(loop, wa, n);
+  wa->cb(loop, wa);
```

Comment:
It's an internal callback.  Barring libuv bugs, it can't be NULL.

Matter of fact, I think there is only ever one `struct uv__async` instance these days, `&loop->async_watcher`, so we can probably avoid the indirection altogether and just call `uv__async_event()` directly.